Sources/StreamChatUI/ChatMessageList/Attachments/ChatFileAttachmentListView+ItemView.swift

```diff
@@ -80,72 +75,64 @@ extension _ChatMessageFileAttachmentListView {
-                actionIconImageView.leadingAnchor.pin(
-                    equalToSystemSpacingAfter: fileNameAndSizeStack.trailingAnchor,
-                    multiplier: 1
-                ),
-                
-                fileNameAndSizeStack.leadingAnchor.pin(
-                    equalToSystemSpacingAfter: fileIconImageView.trailingAnchor,
-                    multiplier: 2
-                ),
-                fileNameAndSizeStack.centerYAnchor.pin(equalTo: centerYAnchor),
-                fileNameAndSizeStack.topAnchor.pin(greaterThanOrEqualTo: layoutMarginsGuide.topAnchor),
-                fileNameAndSizeStack.bottomAnchor.pin(lessThanOrEqualTo: layoutMarginsGuide.bottomAnchor)
-            ])
+            addSubview(mainContainerStackView)
+            mainContainerStackView.pin(to: layoutMarginsGuide)
+            
+            spinnerAndSizeStack.addArrangedSubviews([loadingIndicator, fileSizeLabel])
+            fileNameAndSizeStack.addArrangedSubviews([fileNameLabel, spinnerAndSizeStack])
+            mainContainerStackView.addArrangedSubviews([fileIconImageView, fileNameAndSizeStack, actionIconImageView])
+            
+            spinnerAndSizeStack.axis = .horizontal
+            spinnerAndSizeStack.spacing = 8
+            spinnerAndSizeStack.alignment = .leading
+            
+            fileNameAndSizeStack.axis = .vertical
+            fileNameAndSizeStack.alignment = .leading
+            fileNameAndSizeStack.spacing = 3
+            
+            mainContainerStackView.axis = .horizontal
+            mainContainerStackView.spacing = 8
+            mainContainerStackView.alignment = .center
         }
 
         override open func updateContent() {
             super.updateContent()
 
             fileIconImageView.image = fileIcon
-            fileNameLabel.text = content?.type.rawValue
+            // If we cannot fetch filename, let's use only content type.
+            fileNameLabel.text = content?.payload.title ?? content?.type.rawValue
 
-            if case .uploadingFailed = content?.uploadingState?.state {
+            switch content?.uploadingState?.state {
+            case .uploaded:
+                fileSizeLabel.text = content?.payload.file.sizeString
+            case .uploadingFailed:
                 fileSizeLabel.text = L10n.Message.Sending.attachmentUploadingFailed
-            } else {
+            default:
                 fileSizeLabel.text = content?.uploadingState?.fileUploadingProgress
             }
-
-            if let state = content?.uploadingState?.state {
-                actionIconImageView.image = appearance.fileAttachmentActionIcon(for: state)
-            } else {
+        
+            switch content?.uploadingState?.state {
+            case .none:
                 actionIconImageView.image = nil
+            default:
+                actionIconImageView.image = appearance.fileAttachmentActionIcon(for: content!.uploadingState!.state)
             }
```

Comment:
Wouldn't `if let ...` here work better?

Reply:
yes 😅 reverted... Actually there is one issue, that somehow the content is set back to `nil` and we don't see the `actionIconImageView` when the file is fully uploaded, was taking a look into  this but couldn't find some explanation in a reasonable time, so after I merge this I will take a closer look...